content/posts/open_channel_pre_taproot.md

```diff
@@ -0,0 +1,280 @@
+Once the funding message has been received, Bob can fill in the rest of his
+commitment transaction:
+
+ ![](/openChanPt1/1b_3.png#center)
+
+Alice won’t broadcast the funding transaction until she has a valid signature
+from Bob for her commitment transaction. Enter `funding_signed`:
+
+ ![](/openChanPt1/funding_signed.png#center)
+
+Notice that this is the first message to use the real channel ID instead of the
+temporary one. 
+
+This was the last piece of the puzzle for Alice. She now has all the info she
+needs to be able to sign her commitment transaction if ever needed. 
+
+ ![](/openChanPt1/a1_4.png#center)
+
+Alice can now safely broadcast the funding transaction. Both she and Bob will
+watch the chain for the confirmation of the funding transaction. Once it has
+reached the `minimum_depth` specified by Bob in `accept_channel`, both sides
+will exchange the `channel_ready` message (previously named `funding_locked`).
+This message serves as both a signal to the peer to indicate to them that the
+channel is ready for use (and that the channel announcement process can now
+start if the peers decided on an announced channel) and also to send across each
+peer’s `second_per_commitment_point` that they should use in their second
+commitment transaction.
+
+ ![](/openChanPt1/channel_ready.png#center)
+
+Ok cool! We have completed our first goal: Alice and Bob have opened a channel
+between themselves in a trust-less way. Now we move on to step 2: announcing
+this channel to the network!
+
+## Announcing the channel
+
+This part is fairly painless. Basically there is just one message,
+`channel_announcement`, that Alice and Bob need to construct together and once
+it is complete, then they can broadcast it to the network. Other nodes will use
+this message to prove a few things:
+
+- That the channel funding tx is actually an existing, unspent UTXO with an
+acceptable number of confirmations. 
+- That the funding transaction output actually looks like a lightning channel
+funding transaction
+- That the said channel is actually owned by the keys that Alice and Bob say they
+used to construct the channel.
+- That Alice and Bob both agree on the message being broadcast. 
+
+An incomplete version of the `channel_announcement` message looks as follows:
+
+ ![](/openChanPt1/channel_announcement.png#center)
+
+`h` is the hash of all the data that will be covered by the signatures. In order
+to complete the message, Alice and Bob both compute a signature over `h` using
+the private keys associated with their node IDs and the pubkeys they used in the
+funding transaction. They then both exchange the `announcement_signatures`
+message in order to communicate these signatures to each other:
+
+ ![](/openChanPt1/announcement_sigs.png#center)
+
+Now both nodes can put together the complete `channel_announcement` message:
+
+ ![](/openChanPt1/channel_announcement_2.png#center)
+
+Let’s go over the steps that a node (Charlie) receiving this message will go
```

Comment:
Not sure why charlie needs to verify here? Perhaps my own ignorance, but won't it be super obvious if a channel doesn't exist. I.e. cannot route?

Reply:
Charlie does not want to have to test every channel to know if it is a real channel. He also doesnt want to spread this channel-announcement message to his peers if it is in fact just a spam message.